homeassistant/components/netgear_lte/__init__.py

```diff
@@ -103,6 +117,24 @@ def get_modem_data(self, config):
             hass, cookie_jar=aiohttp.CookieJar(unsafe=True))
         hass.data[DATA_KEY] = LTEData(websession)
 
+        async def delete_sms_handler(service):
+            """Apply a service."""
+            host = service.data[ATTR_HOST]
+            conf = {CONF_HOST: host}
+            modem_data = hass.data[DATA_KEY].get_modem_data(conf)
+
+            if not modem_data:
+                _LOGGER.error(
+                    "%s: host  '%s' unavailable", SERVICE_DELETE_SMS, host)
+                return
+
+            for sms_id in service.data[ATTR_SMS_ID]:
+                await modem_data.modem.delete_sms(sms_id)
```

Comment:
Is call order important? Otherwise we could use `asyncio.wait`.

Reply:
Call order is not important but I am unsure how the modem handles a lot of simultaneous requests. Having a single ID should anyway be more usual so I will just leave it as-is.